File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.cs
using System;

public class Solution 
{
    public string solution(string s, string skip, int index) 
    {
        string result = "";
        foreach(char c in s)
        {
            int count = 0;
            char tmp = c;
            while(count < index)
            {
                int newIndex = (tmp - 'a' + 1) % 26;
                tmp = (char)('a' + newIndex);   // 먼저 이동
                
                if(!skip.Contains(tmp))          // 이동한 후 도착한 곳이 skip이 아니면
                {
                    count++;                      // 그때 카운트
                }
            }
            result += tmp;
        }
        return result;
    }
}

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

string solution(string s, string skip, int index) 
{
    string answer = "";
    vector<char> vec;
    for(int i=0; i<s.length(); i++)
    {
        char c = ++s[i];
        while(vec.size() < index)
        {
            if(c > 'z') { c = 'a'; }
            while(skip.find(c) != string::npos)
            {
                c++;
                if(c > 'z') { c = 'a'; }
            }
            vec.push_back(c);
            c++;
        }
        answer += vec[vec.size() - 1];
        for(char c : vec)
        {
            cout << c << " ";
        }
        vec.clear();
    }
    return answer;
}